File: 14-Longest-Common-Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 1) return strs[0];

        int prefix_length = 0;

        for (int idx = 0; idx < strs[0].size(); idx++) {
            for (int i = 1; i < strs.size(); i++) {
                if (strs[i].empty() || idx >= strs[i].size() || strs[i][idx] != strs[0][idx]) 
                    return strs[0].substr(0, prefix_length);
            }

            prefix_length++;
        }

       return strs[0].substr(0, prefix_length);
    }
};

File: 14-Longest-Common-Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) return \\;

        StringBuilder prefix = new StringBuilder(strs[0]);

        for (int i = 1; i < strs.length; i++) {
            int idx = 0;
            while (idx < prefix.length() && idx < strs[i].length() && strs[i].charAt(idx) == prefix.charAt(idx)) {
                idx++;
            }
            prefix.setLength(idx);

            if (prefix.length() == 0) return \\;
        }

        return prefix.toString();
    }
}